src/hex.ts

```diff
@@ -9,6 +9,10 @@ export const StrictHexStruct = pattern(string(), /^0x[0-9a-f]+$/iu) as Struct<
   Hex,
   null
 >;
+export const HexAddressStruct = pattern(
+  string(),
+  /^0x[0-9a-fA-F]{40}$/u,
```

Comment:
I think that `/^0x[0-9a-fA-F]{40}$/u` is better than `/^0x[0-9a-f]$/u` as other patterns in this file do, otherwise an address that starts with `0X` would be formally correct correct

Reply:
Good catch, that does seem better

Reply:
Nit: could you add a test case for the "0X" case?